src/components/inputs/localized-money-input/localized-money-input.spec.js

```diff
@@ -0,0 +1,629 @@
+
+// We use this component to simulate the whole flow of
+// changing a value and formatting on blur.
+class TestComponent extends React.Component {
+  static displayName = 'TestComponent';
+  static propTypes = {
+    id: PropTypes.string,
+    value: PropTypes.objectOf(PropTypes.string).isRequired,
+    onChange: PropTypes.func,
+    selectedCurrency: PropTypes.string.isRequired,
+  };
+  static defaultProps = {
+    id: 'some-id',
+    name: 'some-name',
+    value: {
+      EUR: '12.77',
+      USD: '13.55',
+      CAD: '19.82',
+    },
+    selectedCurrency: 'CAD',
+    intl: { formatMessage: message => message.id },
+  };
+
+  state = {
+    value: this.props.value || {},
+    selectedCurrency: this.props.selectedCurrency || '',
+  };
+
+  handleChange = event => {
+    event.persist();
+    this.setState({
+      value: {
+        ...this.state.value,
+        [event.target.currency]: event.target.value,
+      },
+    });
+    if (this.props.handleChange) this.props.handleChange(event);
+  };
+
+  render() {
+    return (
+      <LocalizedMoneyInput
+        {...this.props}
+        onChange={this.handleChange}
+        value={this.state.value}
+        selectedCurrency={this.state.selectedCurrency}
+      />
+    );
+  }
+}
+
+const renderLocalizedMoneyInput = (props, options) =>
+  render(<TestComponent {...props} />, options);
+
+it('should forward data-attributes', () => {
+  const { container } = renderLocalizedMoneyInput({
+    'data-foo': 'bar',
+  });
+  expect(container.querySelector('[data-foo="bar"]')).toBeInTheDocument();
+});
+
+it('should have an HTML name', () => {
+  const { container } = renderLocalizedMoneyInput({
+    name: 'foo',
+    selectedCurrency: 'CAD',
+  });
+  expect(
+    container.querySelector('input[name="foo.CAD.amount"]')
+  ).toBeInTheDocument();
+});
+
+it('should call onBlur when input loses focus', () => {
+  const onBlur = jest.fn();
+  const { container } = renderLocalizedMoneyInput({
+    name: 'foo',
+    selectedCurrency: 'CAD',
+    onBlur,
+  });
+  const input = container.querySelector('input[name="foo.CAD.amount"]');
+  input.focus();
+  expect(input).toHaveFocus();
+  input.blur();
+  expect(input).not.toHaveFocus();
+});
+
+describe('when input is collapsed', () => {
+  it('should not show only the `selectedCurrency`', () => {
+    const { container } = renderLocalizedMoneyInput({
+      name: 'foo',
+      selectedCurrency: 'CAD',
+    });
+    expect(
+      container.querySelector('input[name="foo.CAD.amount"]')
+    ).toBeInTheDocument();
+    expect(
+      container.querySelector('input[name="foo.USD.amount"]')
+    ).not.toBeInTheDocument();
+  });
+  it('should allow changing value of the input', () => {
+    const { container } = renderLocalizedMoneyInput({
+      name: 'foo',
+      selectedCurrency: 'CAD',
+    });
+    const event = { target: { value: '17.34' } };
+    const input = container.querySelector('input[name="foo.CAD.amount"]');
```

Comment:
Is there a way to find this input using a label or another approach than `querySelector`? We should strive not to use other getters first.

![image](https://user-images.githubusercontent.com/1765075/50684634-8e0c3380-1016-11e9-9ad9-64c548ebbcc8.png)

See [*"Which get method should I use?"*](https://testing-library.com/docs/faq)